dashboards-observability/public/components/visualizations/charts/stats/stats.tsx

```diff
@@ -0,0 +1,686 @@
+            x: index / metricsLength + ANNOTATION_MARGIN_LEFT,
+            y: 1,
+            metricValue: value,
+            type: 'name',
+          },
+          {
+            ...STATS_ANNOTATION,
+            xanchor: 'left',
+            yanchor: 'top',
+            text: createValueText(value),
+            font: {
+              size: valueSize,
+              color: valueColor,
+              family: 'Roboto',
+            },
+            x: index / metricsLength + ANNOTATION_MARGIN_LEFT,
+            y: 1,
+            type: 'value',
+            metricValue: value,
+          },
+        ]
+      : [
+          {
+            ...STATS_ANNOTATION,
+            x:
+              metricsLength === 1
+                ? 0.5
+                : index === 0
+                ? ((1 / metricsLength) * 1) / 2
+                : (index + 1) / metricsLength - ((1 / metricsLength) * 1) / 2,
+            xanchor: 'center',
+            y: 0.95,
+            yanchor: 'bottom',
+            text: textMode === 'values' ? createValueText(value) : label,
+            font: {
+              size: textMode === 'values' ? valueSize : titleSize,
+              color: textMode === 'names' ? (isDarkMode ? COLOR_WHITE : COLOR_BLACK) : valueColor,
+              family: 'Roboto',
+            },
+            type: textMode === 'names' ? 'name' : 'value',
+            metricValue: value,
+          },
+        ];
+  };
+
+  // extend y axis range to increase height of subplot w.r.t metric data
+  const extendYaxisRange = (metric: ConfigListEntry) => {
+    const sortedData = data[metric.label].slice().sort((curr: number, next: number) => next - curr);
+    const avgSeriesDiff = sortedData.slice(0, 5).reduce(function (prev, curr, index: number) {
+      if (data[metric.label][index + 1])
+        return (prev += Math.abs(Number((data[metric.label][index + 1] - curr).toFixed(2))));
+      return prev;
+    }, 0);
+    return sortedData[0] + (avgSeriesDiff ? avgSeriesDiff : 100);
+  };
+
+  const getMetricValue = (label: string) =>
+    typeof data[label][data[label].length - 1] === 'number'
+      ? getRoundOf(data[label][data[label].length - 1], precisionValue)
+      : 0;
+
+  const generateLineTraces = () => {
+    return metrics.map((metric: ConfigListEntry, metricIndex: number) => {
+      const annotationOption = {
+        label: metric.label,
+        value: getMetricValue(metric.label),
+        index: metricIndex,
+        valueColor: '',
+      };
+      chartLayout = {
+        ...chartLayout,
+        annotations: chartLayout.annotations.concat(
+          orientation === DefaultOrientation || metricsLength === 1
+            ? createAnnotationAutoModeVertical(annotationOption)
+            : createAnnotationsAutoModeHorizontal(annotationOption)
+        ),
+        [`xaxis${metricIndex > 0 ? metricIndex + 1 : ''}`]: {
+          visible: false,
+          showgrid: false,
+          anchor: `y${metricIndex > 0 ? metricIndex + 1 : ''}`,
```

Comment:
`metricIndex > 0 ? metricIndex + 1 : ''` You can put this in a variable and use it everywhere. I see you have been doing this 4 times. We can have just one comparison, if we save it in a variable.